Python/optimizer_bytecodes.c

```diff
@@ -327,10 +327,14 @@ dummy_func(void) {
                 goto error;
             }
             res = sym_new_const(ctx, temp);
+            l = left;
+            r = right;
             Py_DECREF(temp);
         }
         else {
             res = sym_new_type(ctx, &PyUnicode_Type);
+            l = left;
+            r = right;
         }
```

Comment:
This doesn't need to be in the branch. You can write this at the end of the instruction here.

That way, we only need to have one `l = left; r = right`